### PR TITLE
Move towards building with Dune

### PR DESCRIPTION
**Pull request by lindig**
* remove compiler warnings
* port some jbuild files (but some remain to be ported)

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

### CODE REVIEW

File: lib/xenops_server_plugin.ml [hunk 1]
```diff
@@ -50,7 +50,7 @@ let string_of_data x = x |> rpc_of data |> Jsonrpc.to_string
 type flag =
   | Live
 [@@deriving rpcty]
-let string_of_flag x = x |> rpc_of flag |> Jsonrpc.to_string
+let _string_of_flag x = x |> rpc_of flag |> Jsonrpc.to_string
```
Comment (mseri) [inline lib/xenops_server_plugin.ml]: This is shadowed by the definition below, you could just get rid of it

File: Makefile [hunk 2]
```diff
@@ -6,26 +6,26 @@ OPAM_LIBDIR=$(DESTDIR)$(shell opam config var lib)
...
-	jbuilder clean
+	dune clean
 
 release:
-	jbuilder build @install
+	dune build @install --profile=release
 
 test:
-	jbuilder runtest --no-buffer
+	dune runtest --no-buffer
```
Comment (edwintorok) [inline Makefile]: i think we may want `--profile=release` here too.

File: Makefile [hunk 3]
```diff
@@ -6,26 +6,26 @@ OPAM_LIBDIR=$(DESTDIR)$(shell opam config var lib)
...
 
 reindent:
 	git ls-files '*.ml*' '**/*.ml*' | xargs ocp-indent --syntax cstruct -i
 
 #requires odoc
 doc:
-	jbuilder build @doc
+	dune build @doc
```
Comment (edwintorok) [inline Makefile]: --profile=release

File: Makefile [hunk 4]
```diff
@@ -6,26 +6,26 @@ OPAM_LIBDIR=$(DESTDIR)$(shell opam config var lib)
...
 
 install:
-	jbuilder install --prefix=$(OPAM_PREFIX) --libdir=$(OPAM_LIBDIR) xapi-xenopsd
+	dune install --prefix=$(OPAM_PREFIX) --libdir=$(OPAM_LIBDIR) xapi-xenopsd
```
Comment (edwintorok) [inline Makefile]: Does this need --profile=release or `-p` to force it into release mode?

File: Makefile [hunk 5]
```diff
@@ -48,7 +48,7 @@ install:
 	DESTDIR=$(DESTDIR) SBINDIR=$(SBINDIR) QEMU_WRAPPER_DIR=$(QEMU_WRAPPER_DIR) LIBEXECDIR=$(LIBEXECDIR) ETCDIR=$(ETCDIR) ./scripts/make-custom-xenopsd.conf
 
 uninstall:
-	jbuilder uninstall --prefix=$(OPAM_PREFIX) --libdir=$(OPAM_LIBDIR) xapi-xenopsd
+	dune uninstall --prefix=$(OPAM_PREFIX) --libdir=$(OPAM_LIBDIR) xapi-xenopsd
```
Comment (edwintorok) [inline Makefile]: same question as above.

File: lib/xenopsd.ml [hunk 6]
```diff
@@ -171,7 +171,7 @@ let main backend =
       ~rpc_fn
       () in
 
-  (** we need to catch this to make sure at_exit handlers are
+  (* we need to catch this to make sure at_exit handlers are
```
Comment (edwintorok) [inline lib/xenopsd.ml]: This could also be linked to the `signal_handler` by deleting the empty line perhaps.

File: xc/fence/dune [hunk 7]
```diff
@@ -0,0 +1,10 @@
+(executable 
+  (name fence)
+  (public_name fence.bin)
+  (package xapi-xenopsd-xc)
+  (flags -warn-error +a-3)
+  (libraries 
+    xenctrl
+  )
+)
+
```
Comment (edwintorok) [inline xc/fence/dune]: There is now a `dune unstable-fmt` command, do you think it'd be worth testing it on these dune files and submit improvements upstream? (both of us had our own jbuild reformatters)

**Pull request merged**